### PR TITLE
Create individual definition for foundry

### PR DESCRIPTION
**Pull request by anvacaru**
Currently, the generated Haskell definition contains both the semantics for EVM and a new semantics for the Foundry property testing. 
The second one contains rules and productions which are not supposed to apply to the semantics of the EVM under normal circumstances, and which are causing our current proofs for EVM to fail ( see #1412 ).
The changes in this PR generate a new kompiled definition for the `FOUNDRY` module under a `foundry` folder, while the `haskell` will contain only the kompiled semantics which have `EDSL` as the main module.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -278,6 +278,26 @@ $(KEVM_LIB)/$(node_kompiled): $(KEVM_LIB)/$(node_kore) $(protobuf_out) $(libff_o
...
+foundry_main_file     := foundry.md
+foundry_main_filename := $(basename $(notdir $(foundry_main_file)))
+foundry_kompiled_dir  := $(foundry_dir)
+foundry_kompiled      := $(foundry_kompiled_dir)/definition.kore
+
+ifeq ($(UNAME_S),Darwin)
+$(KEVM_LIB)/$(foundry_kompiled): $(libsecp256k1_out)
+endif
+
+$(KEVM_LIB)/$(foundry_kompiled): $(kevm_includes) $(plugin_includes) $(KEVM_BIN)/kevm
```
Comment (ehildenb) [inline Makefile:295]:
```suggestion
$(KEVM_LIB)/$(foundry_kompiled): $(kevm_includes) $(plugin_includes) $(lemma_includes) $(KEVM_BIN)/kevm
```

File: hashed-locations.md [hunk 2]
```diff
@@ -93,6 +93,31 @@ module SOLIDITY-FIELDS
...
+module FOUNDRY-ACCOUNTS
+    imports SOLIDITY-FIELDS
+
+    syntax Contract        ::= FoundryContract
+    syntax Field           ::= FoundryField
+    syntax FoundryContract ::= "Foundry"      [klabel(contract_Foundry)]
+                             | "FoundryTest"  [klabel(contract_FoundryTest)]
+                             | "FoundryCheat" [klabel(contract_FoundryCheat)]
+ // -------------------------------------------------------------------------
+    rule #address(Foundry)      => 137122462167341575662000267002353578582749290296  // 0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38
```
Comment (ehildenb) [inline hashed-locations.md:114]: Why is this moved here? It should be in `foundry.md`.

If the problem is having `Contract` sort defined, only move that definition.

Comment (ehildenb) [reply hashed-locations.md:114]: Same for `Field` sort.

Comment (ehildenb) [reply hashed-locations.md:114]:
```
syntax Contract
syntax Field
```

File: foundry.md [hunk 3]
```diff
@@ -102,6 +102,8 @@ module FOUNDRY-ACCOUNTS
     syntax FoundryContract ::= "Foundry"      [klabel(contract_Foundry)]
                              | "FoundryTest"  [klabel(contract_FoundryTest)]
                              | "FoundryCheat" [klabel(contract_FoundryCheat)]
+    syntax Int ::= #address ( Contract ) [macro]
+ // --------------------------------------------
```
Comment (anvacaru) [inline foundry.md:106]: fix this

**Comment by anvacaru** [comment]
Reopened as #1424.
